lesson 7a comparing architectures

rbavery opened the pull request:
This adds a lesson comparing fundamental developments in deep learning for computer vision, so far up to Resnet. Only two other arcs I plan on adding for this lesson are U-Net/Fully convolutional networks and Vision Transformer but @lillythomas let the know if you think I should add others. I made a separate section for time series and change detection specific architectures, which I think the hugging face/model sourcing section might occur at the end of that to sum up how to find existing models.

Still todo
* finish Unet and ViT descriptions
* improve descriptions of the vanishing gradient and curse of dimensionality to frame Lesson 9b
* time series/remote sensing specific architecture part c with figures and descriptions.

lillythomas commented:
My understanding of dropout is that it works by randomly dropping a number of nodes when passing from one layer to the next to force regularization, not randomly setting weights.

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-142)._

lillythomas commented:
Below are the features learned by the initial layer of alexNet. 
> AlexNet*

Note how some features learn textural information, like lines, and others learn color.
> Would suggest saying "how some nodes, or kernels" rather than features, which to me are the things that are in fact learned.

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-143)._

lillythomas commented:
The amount of parameters is huge given that the last fully connected layers have hundreds of millions of parameters.
> Suggest removing the second mention of parameters.

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-144)._

lillythomas commented:
> since Huggingface has the easiest to use and most well supported API for fetching and using deep learning models.
Capitalize *Since

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-145)._

lillythomas commented:
Use quotations in first mention of curse of dimensionality

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-146)._

lillythomas commented:
Lat bullet under "ConvLSTM and alternatives for time series prediction" is "not all" ?

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-147)._

lillythomas commented:
> "For most of the experiments, the models were designed to keep a computational budget of 1.5 billion multiply-adds at inference time, so that the they do not end up to be a purely academic curiosity, but could be put to real world use, even on large datasets, at a reasonable cost."

Cite.


_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-162)._

lillythomas commented:
> Instead off convolution > max pooling > ReLU, adds a 1x1 convolution to result in convolution ? max pooling > 1x1 convolution > Relu. this is the inception module

Spelling "of", use of question mark is unclear. Capitalization and punctuation last sentence.

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-163)._

lillythomas commented:
> Image Captioning: [Insert: Brief Explanation and Examples]

Incomplete. 

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-164)._

lillythomas commented:
> Generating human images ("This person does not exist"). 

To me, this doesn't land entirely clearly. Maybe rephrase to "Generating fake, yet photo-realistic images by learning from real ones"

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-165)._

lillythomas commented:
> Set and t models outperform spatial models

What is meant by "t" ?

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-166)._

lillythomas commented:
satellit > satellite

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-167)._

lillythomas commented:
> Remove irrelevant changes with normalization, registration,

trailing comma

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-168)._

lillythomas commented:
> Reducing the variability
> Or model the variability, remove with ML, DNNs SVMs

Combine bullets into one

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-169)._

lillythomas commented:
> exemplars

spelling

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-170)._

lillythomas commented:
> EMS-CDNet = uses partition unit for faster feature extraction and merging
> But it struggled with unstrictly registered datasets

Combine.

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-171)._

lillythomas commented:
> M2-CDNET addresses this. Deformable convolution learns offsets between given features to alleviate pixel displacement caused by registration errors. As the scale shrinks…

Last part needs to be completed, not ...

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-172)._

lillythomas commented:
Everything under "Future work NERF, reduce registration errors and projection bias" is incoherent

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-173)._

lillythomas commented:
> Big data Labeling - imbalance, noisy labels Generalization Uncertainty in forecasting Stochastic nature, complex, non-linear interactions between earth system variables Effuse system for fire prediction in Europe assumes pine fuel everywhere

Incoherent

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-174)._

lillythomas commented:
This whole section needs revision re: spelling and thought completion.

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-175)._

lillythomas commented:
This whole section needs revision re: spelling and thought completion.



_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-176)._

lillythomas commented:
This whole section needs revision re: spelling and thought completion.



_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-177)._

lillythomas commented:
@rbavery I am done with my initial review. Overall, I think there is good content here but is largely incoherent and in reed of revision w.r.t. spelling and thought completion (especially the second part). The participants will need an interpretable lesson to revisit outside of the verbal lecture itself. 

Also, these lessons should be `7a` and `7b` respectively, not 9. They were just affiliated with the[ ninth github issue](https://github.com/developmentseed/servir-amazonia-2-internal/issues/9) on the project coordination repo.

lillythomas reviewed (changes requested):
Overall revision w.r.t. spelling and thought completion. Some concepts possibly not defined correctly, such as dropout and GAN image gen. Rename lessons.

rbavery commented:
whoops yes "weights" should be nodes and this should be expanded upon to

"Randomly setting nodes to 0 so that they do not contribute to training.". From https://d2l.ai/chapter_multilayer-perceptrons/dropout.html

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-178)._

rbavery commented:
sorry I should have specified more clearly that Lesson 9b is what I needed reviewed, 9c is mostly notes that are still getting refactored. I think given the Session8/Lesson 10 content, some will go to that lesson before hyperparam tuning and some to 9b. 9c will mostly just be for demonstrating hugging face

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-187)._

rbavery commented:
sorry should have specified this notes don't need review.

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-188)._

rbavery commented:
Sorryy I should have elaborated in the GitHub description that 9c wasn't ready for review, it's still a TODO. Lesson 9b is ready to be included now, I've addressed your comments there and removed 9c from this PR

lillythomas commented:
Still describes dropout as "reduces overfitting by randomly setting learned weights" so I defer to my earlier comment about this.

_Respond and view the context [here](https://gitnotebooks.com/developmentseed/tensorflow-eo-training-2/pull/7#comment-191)._
